### PR TITLE
SW-18: Redundant link manager/switcher

### PR DESCRIPTION
**Pull request by robertlong13**
https://carbonix.atlassian.net/browse/SW-18

### CODE REVIEW

File: plugins/RedundantLinkManager/RedundantLinkManager.cs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+{
+    public partial class RedundantLinkManager : Form
+    {
+        public RedundantLinkManager()
+        {
+            InitializeComponent();
+        }
+
+        // Force the "Priority" column to match the row index when reordering
+        private void myDataGridView1_RowPostPaint(object sender, DataGridViewRowPostPaintEventArgs e)
```
Comment (Lokesh-Carbonix) [inline plugins/RedundantLinkManager/RedundantLinkManager.cs]: just correct the naming myDataGridView1_RowPostPaint

File: Common.cs [hunk 2]
```diff
@@ -63,6 +65,7 @@ public static GMapMarker getMAVMarker(MAVState MAV, GMapOverlay overlay = null)
                         itemr.Cog = MAV.cs.groundcourse;
                         itemr.Target = MAV.cs.target_bearing;
                         itemr.Nav_bearing = MAV.cs.nav_bearing;
+                        itemr.IsActive = MainV2.comPort.MAV == MAV;
```
Comment (Pradeep-Carbonix) [inline Common.cs:68]: Is this required ?

File: ExtLibs/Maps/GMapMarkerBoat.cs [hunk 3]
```diff
@@ -69,9 +69,26 @@ public override void OnRender(IGraphics g)
             {
             }
 
+            // Draw the boat with transparency (if possible)
+#if NET472_OR_GREATER
+            var colorMatrix = new System.Drawing.Imaging.ColorMatrix();
+            if(!IsActive)
+            {
+                colorMatrix.Matrix33 = 0.39f;
```
Comment (Pradeep-Carbonix) [inline ExtLibs/Maps/GMapMarkerBoat.cs:77]: no default app resources color ?

File: ExtLibs/Maps/GMapMarkerPlane.cs [hunk 4]
```diff
@@ -165,20 +165,30 @@ public override void OnRender(IGraphics g)
             // the plane
             g.TranslateTransform(-2, -2);
 
+            var fill_color = Color.White;
```
Comment (Pradeep-Carbonix) [inline ExtLibs/Maps/GMapMarkerPlane.cs:168]: white ?

File: MissionPlanner.sln [hunk 5]
```diff
@@ -891,6 +893,14 @@ Global
 		{0B99EBF2-B964-4AAC-9EC5-76353610AAAE}.Release|Any CPU.Build.0 = Release|Any CPU
 		{0B99EBF2-B964-4AAC-9EC5-76353610AAAE}.Release|x86.ActiveCfg = Release|Any CPU
 		{0B99EBF2-B964-4AAC-9EC5-76353610AAAE}.Release|x86.Build.0 = Release|Any CPU
+		{6B6AF209-F65C-41A5-8B70-A60067EF9AF3}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
+		{6B6AF209-F65C-41A5-8B70-A60067EF9AF3}.Debug|Any CPU.Build.0 = Debug|Any CPU
+		{6B6AF209-F65C-41A5-8B70-A60067EF9AF3}.Debug|x86.ActiveCfg = Debug|Any CPU
+		{6B6AF209-F65C-41A5-8B70-A60067EF9AF3}.Debug|x86.Build.0 = Debug|Any CPU
+		{6B6AF209-F65C-41A5-8B70-A60067EF9AF3}.Release|Any CPU.ActiveCfg = Release|Any CPU
+		{6B6AF209-F65C-41A5-8B70-A60067EF9AF3}.Release|Any CPU.Build.0 = Release|Any CPU
+		{6B6AF209-F65C-41A5-8B70-A60067EF9AF3}.Release|x86.ActiveCfg = Release|Any CPU
```
Comment (Pradeep-Carbonix) [inline MissionPlanner.sln:902]: Repeated ?
`A60067EF9AF3}.Release|Any CPU.ActiveCfg = Release|Any CPU
		{6B6AF209-F65C-41A5-8B70-A60067EF9AF3}.Release|Any CPU.Build.0 = Release|Any CPU
		{6B6AF209-F65C-41A5-8B70-A60067EF9AF3}.Release|x86.ActiveCfg = Release|Any CPU`

File: ExtLibs/Maps/GMapMarkerPlane.cs (hunk 4 above)
Comment (robertlong13) [reply ExtLibs/Maps/GMapMarkerPlane.cs:168]: You're right, this should be cleaned up. Basically, this initializer is unnecessary, as it is guaranteed to get overwritten in the next few lines. it's just there to shut up the code analysis from complaining that fill_color may be uninitialized (even though it's guaranteed to be).

I'll clean this up in my upstream PR. I'll leave this as is here for now.

File: plugins/RedundantLinkManager/Link.cs [hunk 6]
```diff
@@ -0,0 +1,330 @@
...
+
+        private string _type;
+        public string Type
+        {
+            get { return _type; }
+            set
+            {
+                if (_type == value) return;
+                _type = value;
+                Dispose();
```
Comment (Pradeep-Carbonix) [inline plugins/RedundantLinkManager/Link.cs:34]: why dispose here ?

File: ExtLibs/Maps/GMapMarkerBoat.cs (hunk 3 above)
Comment (robertlong13) [reply ExtLibs/Maps/GMapMarkerBoat.cs:77]: All the GMapMarker implementations are a bit of a hardcoded nightmare. Handling this transparency value in a resource would be the right way to do it, but nothing else in them is being handled this way yet. They need a heavy refactoring.

File: Common.cs (hunk 2 above)
Comment (robertlong13) [reply Common.cs:68]: Not required for us, but part of my upstream PR.

File: plugins/RedundantLinkManager/Link.cs (hunk 6 above)
Comment (robertlong13) [reply plugins/RedundantLinkManager/Link.cs:34]: These Link objects are bound to the DataGridView rows in the manager UI. If any of the settings change that would require the link to be closed and reopened, I need to dispose.

**Review by Pradeep-Carbonix — APPROVED** [review]

**Comment by robertlong13** [comment]
I've done a rebase update

**Comment by robertlong13** [comment]
I have updated this to be disabled by default. Give that enable/disable feature a test and then I think we should merge.

**Pull request merged**